lib/api/CDataFrameAnalyzer.cc

```diff
@@ -153,7 +157,15 @@ void CDataFrameAnalyzer::run() {
     auto outStream = m_OutStreamSupplier();
     core::CRapidJsonConcurrentLineWriter outputWriter{*outStream};
 
-    this->monitorProgress(*analysis, outputWriter);
+    auto persistStream = m_PersistStreamSupplier();
+    if (persistStream.get() == nullptr) {
+        HANDLE_FATAL(<< "Input error: File for state persistence has to be specified.");
```

Comment:
I know we discussed this briefly, but I think this is a pain for debugging. Running from the Java it should always be there, but from the command line I think this would be optional.

Reply:
This has become obsolete, since we are not operating with persistStreamSupplier here. In `CDataFrameAnalysisSpecification` I introduced a noop to handle the default argument option.